src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,121 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.models.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    private Connection connection = ConnectionUtil.getConnection();
+
+    @Override
+    public List<Manufacturer> getAll() {
+        List<Manufacturer> allManufacturers = new ArrayList<>();
+        String getAllManufacturerRequest
+                = "SELECT * FROM manufacturers WHERE is_deleted = false";
+        try (PreparedStatement getAllManufacturerStatement = connection
+                .prepareStatement(getAllManufacturerRequest)) {
+            ResultSet resultSet
+                    = getAllManufacturerStatement.executeQuery();
+            while (resultSet.next()) {
+                allManufacturers.add(getManufacturer(resultSet));
+            }
+            return allManufacturers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all manufacturers "
+                    + "from manufacturers table", e);
+        }
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateRequest = "UPDATE manufacturers SET name = ?, country = ? "
+                + "WHERE id = ? AND is_deleted = false;";
+        try (PreparedStatement updateManufacturerStatement = connection
+                .prepareStatement(updateRequest, Statement.RETURN_GENERATED_KEYS)) {
+            updateManufacturerStatement.setString(1, manufacturer.getName());
+            updateManufacturerStatement.setString(2, manufacturer.getCountry());
+            updateManufacturerStatement.setLong(3, manufacturer.getId());
+            updateManufacturerStatement.executeUpdate();
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacturer to DB", e);
+        }
```

Comment:
let's make our error messages more informative, do this everywhere 
```suggestion
            throw new DataProcessingException("Can't update manufacturer to DB " + manufacturer, e);
        }
```